pkg/agent/flannel/setup.go

```diff
@@ -218,16 +218,6 @@ func createFlannelConf(nodeConfig *config.Node) error {
 
 	var backendConf string
 
-	// precheck and error out unsupported flannel backends.
-	switch nodeConfig.Flannel.Backend {
-	case BackendHostGW:
-	case BackendTailscale:
-	case BackendWireguardNative:
-		if goruntime.GOOS == "windows" {
-			return fmt.Errorf("unsupported flannel backend '%s' for Windows", nodeConfig.Flannel.Backend)
```

Comment:
This logic is not longer needed? The precheck seemed valuable to shortciruit if windows was detected.

Reply:
@manuelbuil was the idea here for all three backends to be unsupported? golang switch cases do not fall through by default, so this was only erroring out on wireguard - the first two cases are no-ops.

Reply:
I am going to assume this was a bug, and all three cases were meant to fall-through and return unsupported. I have refactored to observe this behavior.